vaadin-combo-box-flow-parent/vaadin-combo-box-flow/src/test/java/com/vaadin/flow/component/combobox/ComboBoxBaseTest.java

```diff
@@ -295,7 +295,7 @@ public void setDataProvider_backendDataProvider_fetchesOnOpened() {
         DataProvider<String, Void> dataProvider = Mockito.spy(DataProvider
                 .fromCallbacks(query -> Stream.empty(), query -> 0));
 
-        comboBox.setDataProvider(dataProvider, filter -> null);
+        comboBox.setItems((DataProvider) dataProvider);
```

Comment:
I wonder if it would be better to make `dataProvider` of type `DataProvider<String, String>` (e.g. using `DataProvider.fromFilteringCallbacks`) and avoid the unchecked cast

Reply:
Good point.